QueryBuilder/Dynamic/JoinQuery.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.DigitalWorkplace.DigitalTwins.QueryBuilder.Dynamic
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.DigitalWorkplace.DigitalTwins.QueryBuilder.Common.Clauses;
+    using Microsoft.DigitalWorkplace.DigitalTwins.QueryBuilder.Common.Statements;
+
+    /// <summary>
+    /// A wrapper for the common functionalities of query builder.
+    /// </summary>
+    public abstract class JoinQuery<TQuery, TWhereStatement> : FilteredQuery<TQuery, TWhereStatement>
```

Comment:
Personally, I'd suggest the opposite if we're being "nit" picky. :) I think _FilterQuery_ is better than _FilteredQuery_. Filter**ed** makes me thing it might be some processed Queryable result.